consensus/src/ordered_broadcast/types.rs

```diff
@@ -0,0 +1,782 @@
+}
+
+/// Node is a message from a sequencer that contains a Chunk and a proof that the parent was correctly broadcasted.
+///
+/// It represents a newly-proposed tip of the chain for the given sequencer. The node includes:
+/// 1. The chunk itself (sequencer, height, payload)
+/// 2. The sequencer's signature over the chunk
+/// 3. For non-genesis nodes (height > 0), proof that the previous chunk was acknowledged by a quorum of validators
+///
+/// Nodes form a linked chain from each sequencer, ensuring that new chunks can only be added
+/// after their predecessors have been properly acknowledged by the validator set.
+#[derive(Clone, Debug)]
+pub struct Node<C: Verifier, D: Digest> {
+    /// Chunk of the node.
+    pub chunk: Chunk<C::PublicKey, D>,
+
+    /// Signature of the sequencer over the chunk.
+    pub signature: C::Signature,
+
+    /// Information about the parent chunk (previous height)
+    ///
+    /// This part is not signed over, but it is used to verify that the previous chunk
+    /// in the chain was correctly broadcast. It contains the threshold signature that
+    /// proves a quorum of validators acknowledged the parent.
+    ///
+    /// For genesis nodes (height = 0), this is None.
+    pub parent: Option<Parent<D>>,
+}
+
+impl<C: Verifier, D: Digest> Node<C, D> {
+    /// Create a new node with the given chunk, signature, and parent.
+    ///
+    /// For genesis nodes (height = 0), parent should be None.
+    /// For all other nodes, parent must be provided.
+    pub fn new(
+        chunk: Chunk<C::PublicKey, D>,
+        signature: C::Signature,
+        parent: Option<Parent<D>>,
+    ) -> Self {
+        Self {
+            chunk,
+            signature,
+            parent,
+        }
+    }
+
+    /// Verify the Node (and its parent).
+    ///
+    /// This ensures:
+    /// 1. The sequencer's signature over the chunk is valid
+    /// 2. For non-genesis nodes, the parent's threshold signature is valid
+    ///
+    /// If verification is successful, returns:
+    /// - None for genesis nodes
+    /// - Some(parent_chunk) for non-genesis nodes
+    ///
+    /// If verification fails, returns an appropriate error.
+    pub fn verify(
+        &self,
+        namespace: &[u8],
+        public: Option<&Public>,
+    ) -> Result<Option<Chunk<C::PublicKey, D>>, Error> {
+        // Verify chunk
+        let chunk_namespace = chunk_namespace(namespace);
+        let message = self.chunk.encode();
+        if !C::verify(
+            Some(chunk_namespace.as_ref()),
+            &message,
+            &self.chunk.sequencer,
+            &self.signature,
+        ) {
+            return Err(Error::InvalidSequencerSignature);
+        }
+        let Some(parent) = &self.parent else {
+            return Ok(None);
+        };
+
+        // Verify parent (if present)
+        let Some(public) = public else {
+            unreachable!("public should always be present when parent is present");
```

Comment:
Nit: add a `#Panics` section to the rustdoc for this function or return an error here